### PR TITLE
Request to merge Complete list

### PR DESCRIPTION
**Pull request by Kingstalux**
For the final milestone of this project, I implemented the CRUD (create, update, delete) methods.
Users can add new tasks by clicking the enter icon or pressing the enter key after writing the task description.
Users can edit tasks by click on the icon on each task, editing the description, and pressing enter.
Users can remove tasks by clicking on the delete icon.
Users can clear the completed tasks by clicking on the button. 
All the elements of the user interface are fully functional and the application is completed.

### CODE REVIEW

File: src/script2.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const newItem = document.querySelector('.new-item').value;
+  const i = taskList.length;
+  const element = {
+    description: newItem,
+    completed: false,
+    index: i,
+  };
+  taskList.push(element);
+  display(element);
+}
```
Comment (EricMbouwe) [inline src/script2.js]: - [x] Tasks added to the todo list should have 1-based indexes, which means that the property `index` should start from `1`.
(i.e. if a 5th task is added to the list, the index of that task should equal to 5). Currently, it starts from `0`, please go ahead and fix that.

Screenshot from 2021-09-30 19-21-06

File: src/script2.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  });
+}
+
+export function clearCompleted(taskList) {
+  taskList = taskList.filter((element) => {
+    if (element.completed !== true) {
+      return true;
+    }
+    return false;
+  });
```
Comment (EricMbouwe) [inline src/script2.js]: - [OPTIONAL] You can refactor this to a shorter version
```
taskList = taskList.filter((element) => element.completed !== true);
```

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Kingstalux, 🖐

Good job so far! 💪 💪 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/script2.js (hunk 1 above)
Comment (Kingstalux) [reply src/script2.js]: okay, let me correct that.

File: src/script2.js (hunk 2 above)
Comment (Kingstalux) [reply src/script2.js]: okay, thanks for the shorter version, I'll apply it.

File: src/script2.js (hunk 1 above)
Comment (rloterh) [reply src/script2.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Kingsely, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**